server/app/controllers/admin/AdminApplicationController.java

```diff
@@ -151,13 +155,39 @@ public Result downloadDemographics() {
   /** Download a PDF file of the application to the program. This feature is not implemented yet. */
   @Secure(authorizers = Authorizers.Labels.ANY_ADMIN)
   public Result download(Http.Request request, long programId, long applicationId) {
+    ProgramDefinition program;
     try {
-      ProgramDefinition program = programService.getProgramDefinition(programId);
+      program = programService.getProgramDefinition(programId);
       checkProgramAdminAuthorization(profileUtils, request, program.adminName()).join();
-      throw new UnsupportedOperationException("Not yet implemented.");
     } catch (ProgramNotFoundException e) {
       return notFound(e.toString());
+    } catch (CompletionException e) {
+      return unauthorized();
+    }
+
+    Optional<Application> applicationMaybe =
+            this.applicationRepository.getApplication(applicationId).toCompletableFuture().join();
+
+    if (!applicationMaybe.isPresent()) {
+      return notFound(String.format("Application %d does not exist.", applicationId));
+    }
+
+    Application application = applicationMaybe.get();
+    String applicantNameWithApplicationId =
+            String.format("%s (%d)", application.getApplicantData().getApplicantName(), application.id);
+
+    String filename = String.format("%s-%s.pdf", applicantNameWithApplicationId, clock.instant().toString());
+    byte[] pdf = null;
+    try{
+      pdf = pdfExporter.export(application);
+    }
+    catch(Exception e)
```

Comment:
issue (blocking): Catching all exceptions with `catch (Exception e)` is pretty risky -- there are a number of reasons the action could fail and with this code in place they would all result in a 404, which would mask server errors in monitoring and alerting tools.

Please change this to only catching errors that require explicit handling. If you are unsure how to handle something going wrong on the server, most likely the best thing to do is to catch the exception and rethrow it wrapped in a `RuntimeException`, which are [unchecked](https://docs.oracle.com/javase/7/docs/api/java/lang/RuntimeException.html).